cmd/gazelle/update-repos.go

```diff
@@ -481,7 +481,8 @@ func ensureMacroInWorkspace(uc *updateReposConfig, insertIndex int) (updated boo
 	// be called somewhere else.
 	macroValue := uc.macroFileName + "%" + uc.macroDefName
 	for _, d := range uc.workspace.Directives {
-		if d.Key == "repository_macro" && d.Value == macroValue {
+		m, defName, _, _ := repo.ParseRepositoryMacroDirective(d.Value)
```

Comment:
We should avoid calling `ParseRepositoryMacroDirective` unless `d.Key == "repository_macro"` (sadly leading to an extra nested `if`):
```
if d.Key == "repository_macro" {
   if m, defName, _, _ := repo. ParseRepositoryMacroDirective(d.Value); m == uc.macroFileName && defName == uc.macroDefName
}
```